.gitlab/dev_container_deploy/docker_linux.yml

```
@@ -287,15 +287,14 @@ qa_agent_ot:
   extends: .docker_publish_job_definition
   stage: dev_container_deploy
   rules:
-    # incident-28463: this job is currently broken
-    # - !reference [.on_container_or_e2e_changes]
+    - !reference [.on_container_or_e2e_changes]
```

<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Do we still want to run it on containers code changes? Does not looks like a good path fit. Would rather have a dedicated rule for otel code

---

Good point.
I don’t know.
So far, it seems that [this `qa_agent_ot` job isn’t a dependency of any other job](https://github.com/search?q=repo%3ADataDog%2Fdatadog-agent%20qa_agent_ot&type=code).
So, there is no technical constraint for the rules about when this job should be triggered.

Given the name of the target, I imagine that this image will eventually be consumed by OTEL e2e tests.
In such a case, a `.on_otel_or_e2e_changes` might make more sense.

So… ❓ question for the original author @truthbk : should the agent `-ot` docker image be pushed on the AWS `agent-qa` project used by e2e tests when files owned by @DataDog/container-integrations are touched or should we keep this job as only manual ?

---

agree it needs to be added in https://github.com/DataDog/datadog-agent/blob/9b1b10cea86c23ca6f530d4d849ee99efbbd5387/.gitlab-ci.yml#L825 

however maybe for to now unblock the PR it can use the `.on_e2e_main_release_or_rc` that is the parent of any `.on*` rule

```suggestion
    - !reference [.on_e2e_main_release_or_rc]
```